### PR TITLE
re-enable the test that's mysteriously failing in .Net core

### PR DESCRIPTION
**Pull request by AnthonySteele**
but only for windows builds
Made a test category "NotInNetCore" for this.

### CODE REVIEW

File: src/NLog.StructuredLogging.Json.Tests/EndToEnd/ViaLayout/WithFailingLayout.cs [hunk 1]
```diff
@@ -57,7 +57,7 @@ public void ShouldSurviveWhenLayoutFails()
         }
 
         [Test]
-        [Ignore("todo: investigate why we don't get this failing prop out when in .Net core")]
+        [Category("NotInNetCore")]
```
Comment (martincostello) [inline src/NLog.StructuredLogging.Json.Tests/EndToEnd/ViaLayout/WithFailingLayout.cs:60]: Maybe "Not Linux"? I thought it worked in AppVeyor for .NET Core?

Comment (AnthonySteele) [reply src/NLog.StructuredLogging.Json.Tests/EndToEnd/ViaLayout/WithFailingLayout.cs:60]: I don't know if it's caused by Linux or by .NET Core. We don't have a .net core test run in Appveyor.

Comment (martincostello) [reply src/NLog.StructuredLogging.Json.Tests/EndToEnd/ViaLayout/WithFailingLayout.cs:60]: You're right. Sorry, I thought we were using ```dotnet test``` there too.

**Comment by AnthonySteele** [comment]
https://github.com/NLog/NLog/issues/2057

**Comment by AnthonySteele** [comment]
Should I rename the category, or can we merge?

**Pull request merged**